pysat/coords.py

```diff
@@ -0,0 +1,263 @@
+import scipy
+import scipy.integrate
+import numpy as np
+
+
+def geocentric_to_ecef(latitude, longitude, altitude):
+    """Convert geocentric coordinates into ECEF
+    
+    Parameters
+    ----------
+    latitude : float or array_like
+        Geocentric latitude (degrees)
+    longitude : float or array_like
+        Geocentric longitude (degrees)
+    altitude : float or array_like
+        Height (km) above presumed spherical Earth with radius 6371 km.
+        
+    Returns
+    -------
+    x, y, z
+        numpy arrays of x, y, z locations in km
+        
+    """
+
+    r = 6371. + altitude
+    # colatitude = 90. - latitude
+    x = r * np.cos(np.deg2rad(latitude)) * np.cos(np.deg2rad(longitude))
+    y = r * np.cos(np.deg2rad(latitude)) * np.sin(np.deg2rad(longitude))
+    z = r * np.sin(np.deg2rad(latitude))
+
+    return x, y, z
+
+
+def ecef_to_geocentric(x, y, z, ref_height=6371.):
+    """Convert ECEF into geocentric coordinates
+    
+    Parameters
+    ----------
+    x : float or array_like
+        ECEF-X in km
+    y : float or array_like
+        ECEF-Y in km
+    z : float or array_like
+        ECEF-Z in km
+    ref_height : float or array_like
+        Reference radius used for calculating height
+    Returns
+    -------
+    latitude, longitude, altitude
+        numpy arrays of locations in degrees, degrees, and km
+        
+    """
+
+    r = np.sqrt(x ** 2 + y ** 2 + z ** 2)
+    colatitude = np.rad2deg(np.arccos(z / r))
+    longitude = np.rad2deg(np.arctan2(y, x))
+    latitude = 90. - colatitude
+
+    return latitude, longitude, r - ref_height
+
+
+def geodetic_to_ecef(latitude, longitude, altitude):
+    """Convert WGS84 geodetic coordinates into ECEF
+    
+    Parameters
+    ----------
+    latitude : float or array_like
+        Geodetic latitude (degrees)
+    longitude : float or array_like
+        Geodetic longitude (degrees)
+    altitude : float or array_like
+        Geodetic Height (km) above WGS84 reference ellipsoid.
+        
+    Returns
+    -------
+    x, y, z
+        numpy arrays of x, y, z locations in km
+        
+    """
+
+    b = 6356.75231424518
```

Comment:
Comments stating what a and b are would be good.  Also, could have these defined up top?